### PR TITLE
add pytest run to conda build

### PR DESCRIPTION
**Pull request by CagtayFabry**
## Changes
Add default pytest run to the conda build step

also updated the local recipe to match the conda-force recipe more closely
we should add test run to the conda-forge recipe as well once the setup is refined

### CODE REVIEW

**Comment by marscher** [comment]
Actually you want to copy the test directory during the test phase of conda build. Have a look here https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#test-section

File: conda.recipe/meta.yaml [hunk 1]
```diff
@@ -1,34 +1,74 @@
...
 
 test:
-    imports:
-        - weldx
-        - weldx.visualization
+  requires:
+    - pytest
+    - pytest-xdist
+  source_files:
+    - tests/*.py
```
Comment (marscher) [inline conda.recipe/meta.yaml:32]: Does passing "tests/" here not work?

Why do you need to pass the icons separately? How are they used in the tests?

**Comment by marscher** [comment]
because of #320 you could directly use the test command "pytest --pyargs tests" but I guess we do not really want that.

File: conda.recipe/meta.yaml (hunk 1 above)
Comment (CagtayFabry) [reply conda.recipe/meta.yaml:32]: they get serialized as external files in a part of the asdf test suite

https://github.com/BAMWelDX/weldx/blob/bd643850b670a9da087e7739406bc7074a0dcc84/tests/asdf_tests/test_asdf_core.py#L416-L430

**Review by vhirtham — APPROVED** [review]

File: conda.recipe/meta.yaml [hunk 2]
```diff
@@ -1,34 +1,74 @@
...
+    comparability and reproducibility across different experimental setups can
+    be improved.
+  doc_url: https://weldx.readthedocs.io
+  dev_url: https://github.com/BAMWelDX/weldx/
+
+extra:
+  recipe-maintainers:
+    - marscher
+    - vhirtham
+    - CagtayFabry
```
Comment (vhirtham) [inline conda.recipe/meta.yaml:73]: Did you put your name on purpose to the bottom of the list? 😛

**Review by marscher — APPROVED** [review]
LGTM

File: conda.recipe/meta.yaml (hunk 2 above)
Comment (CagtayFabry) [reply conda.recipe/meta.yaml:73]: actually that was @marscher 😉

Comment (marscher) [reply conda.recipe/meta.yaml:73]: Well it is not an authorship list of a nature paper. I admit I haven't spent any thought on the order.

File: conda.recipe/meta.yaml [hunk 3]
```diff
@@ -1,34 +1,69 @@
...
-        - weldx
-        - weldx.visualization
+  requires:
+    - pytest
+    - pytest-xdist
+  imports:
+    - weldx
+    - weldx.visualization
+  commands:
+    - pytest --pyargs weldx.tests -n auto
```
Comment (CagtayFabry) [inline conda.recipe/meta.yaml:35]: this looks much cleaner now after #323 👍

Comment (marscher) [reply conda.recipe/meta.yaml:35]: yes, indeed - very simple :)
And we are testing the packaged code, not the pointer to the working copy, where everything is always in the right spot. This is another benefit.

**Review by marscher — APPROVED** [review]

File: conda.recipe/meta.yaml (hunk 2 above)
Comment (vhirtham) [reply conda.recipe/meta.yaml:68]: I suspected that Cagtay doesn't want to be bothered with maintenance questions 🤣

**Pull request merged**